ostd/src/mm/heap_allocator.rs

```diff
@@ -78,12 +78,17 @@ unsafe impl<const ORDER: usize> GlobalAlloc for LockedHeapWithRescue<ORDER> {
             return core::ptr::null_mut::<u8>();
         }
 
-        self.heap
+        // We create and return the result to prevent the race condition.
+        //
+        // See issue: <https://github.com/asterinas/asterinas/issues/1085>
+        let res = self
+            .heap
```

Comment:
Why does this change prevent race conditions? It seems that the new version is equivalent to the old one in any meaningful way.

Reply:
The generated code is different and I haven't discovered what makes the race condition disappear.

Reply:
I am ok with the idea of accepting this change as a temporary workaround. But the comments must be removed or reworded

```rust
        // We create and return the result to prevent the race condition.
        //
        // See issue: <https://github.com/asterinas/asterinas/issues/1085>
```

as it seems to suggest that the problem is solved. But in theory, the new version is no different from the old one. So whatever the root cause is, I don't think the problem is really solved.